### ISSUE TITLE
AWS MQ broker security group updates should update in-place instead of replace the broker

### ISSUE THREAD

**Issue by rabidscorpio** [terraform-providers/terraform-provider-aws#10209]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.12.8

### Affected Resource(s)

* aws_mq_broker

### Expected Behavior

Updating the security group for an `aws_mq_broker` will update the resource instead of replacing it.

### Actual Behavior

``` 
# module.staging_data_stores.aws_mq_broker.main must be replaced
.
.
security_groups            = [
          - "sg-xxxxxxxxxxxx",
        ] -> (known after apply) # forces replacement
```

### Important Factoids

As of Aug 30th, you can update the security groups of an AWS MQ broker without replacing the broker.

### References

* https://docs.aws.amazon.com/amazon-mq/latest/developer-guide/amazon-mq-release-notes.html

**Comment by albertoal** [comment]
Hey everyone, I'd like to work on this enhancement and will submit a PR soon. We actually requested this feature to the AWS team as it was causing us issues at https://github.com/BenchLabs/ 

I will link the PR to this issue when I submit it.

**Comment by bflad** [comment]
Support for updating in place has been merged and will release with version 2.35.0 of the Terraform AWS Provider, Thursday next week. Thanks to @albertoal for the implementation. 👍
